pallets/chain-space/src/tests.rs

```diff
@@ -1126,3 +1126,45 @@ fn add_delegate_should_fail_if_space_delegates_limit_exceeded() {
 		);
 	});
 }
+
+#[test]
+fn add_delegate_should_fail_if_delegate_not_found() {
+	let creator = DID_00;
+	let author = ACCOUNT_00;
+	let space = [2u8; 256].to_vec();
+	let capacity = 3u64;
+	let space_digest = <Test as frame_system::Config>::Hashing::hash(&space.encode()[..]);
+
+	let id_digest = <Test as frame_system::Config>::Hashing::hash(
+		&[&space_digest.encode()[..], &creator.encode()[..]].concat()[..],
+	);
+
+	let space_id: SpaceIdOf = generate_space_id::<Test>(&id_digest);
+
+	let auth_id_digest = <Test as frame_system::Config>::Hashing::hash(
+		&[&space_id.encode()[..], &creator.encode()[..]].concat()[..],
+	);
+
+	let authorization_id: AuthorizationIdOf = generate_authorization_id::<Test>(&auth_id_digest);
+
+	new_test_ext().execute_with(|| {
+		assert_ok!(Space::create(
+			DoubleOrigin(author.clone(), creator.clone()).into(),
+			space_digest,
+		));
+
+		assert_ok!(Space::approve(RawOrigin::Root.into(), space_id.clone(), capacity));
+
+		// Attempt to add a delegate that does not exist
+		assert_err!(
```

Comment:
`DelegateNotFound` error is specific to `remove_delegate` and not on `add_delegate`.
Steps to test this:
- Remove a delegate which does not exist(delegate not added and not present in chain as a delegate).